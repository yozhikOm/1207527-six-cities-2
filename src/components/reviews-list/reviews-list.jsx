import React from 'react';
import PropTypes from 'prop-types';
import {Review} from '../review/review.jsx';
import {ReviewForm} from '../review-form/review-form.js';

const ReviewsList = (props) => {
  const {isAuthorizationRequired, offerId, reviews, postReview} = props;

  return (
    <section className="property__reviews reviews">
      <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>
      <ul className="reviews__list">
        {reviews.map((item) => (
          <React.Fragment key={item.id}>
            <Review review={item} />
          </React.Fragment>
        ))}
      </ul>
      {isAuthorizationRequired ? <React.Fragment /> : <ReviewForm offerId={offerId} postReview={postReview}/>}
    </section>

  );
};

ReviewsList.propTypes = {
  isAuthorizationRequired: PropTypes.bool,
  reviews: PropTypes.array,
  offerId: PropTypes.number,
  postReview: PropTypes.func,
};

export {ReviewsList};
