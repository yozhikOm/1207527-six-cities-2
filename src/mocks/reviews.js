export const reviews = [
  {
    id: 1,
    user: {
      id: 1,
      isPro: true,
      name: `Erika`,
      avatarUrl: ``,
    },
    rating: 4.6,
    comment: `review comment`,
    date: `Июль 2018`,
  },

];
