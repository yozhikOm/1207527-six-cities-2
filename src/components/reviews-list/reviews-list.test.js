import React from 'react';
import {ReviewsList} from './reviews-list.jsx';
import ShallowRenderer from 'react-test-renderer/shallow';
import {reviews} from '../../mocks/reviews.js';

it(`ReviewsList компонент рендерится корректно`, () => {
  const renderer = new ShallowRenderer();

  renderer.render(<ReviewsList
    isAuthorizationRequired={false}
    offerId={1}
    reviews={reviews}
    postReview={jest.fn()}
  />);

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
