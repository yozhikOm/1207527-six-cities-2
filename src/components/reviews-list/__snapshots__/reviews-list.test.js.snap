// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsList компонент рендерится корректно 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <React.Fragment>
      <Review
        review={
          Object {
            "comment": "review comment",
            "date": "Июль 2018",
            "id": 1,
            "rating": 4.6,
            "user": Object {
              "avatarUrl": "",
              "id": 1,
              "isPro": true,
              "name": "Erika",
            },
          }
        }
      />
    </React.Fragment>
  </ul>
  <ReviewForm
    offerId={1}
    postReview={[MockFunction]}
  />
</section>
`;
